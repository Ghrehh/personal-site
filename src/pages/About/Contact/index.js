import React from 'react';

import Heading from 'components/Heading';
import Button from 'components/Button';
import styles from './styles.module.css';

const buttonStyle = additionalStyle => `${styles.button} ${additionalStyle}`;

const Contact = () => (
  <section>
    <Heading className={styles.heading}>CONTACT</Heading>

    <div className={styles.buttonContainer}>
      <Button className={buttonStyle(styles.twitterButton)}>Twitter</Button>

      <Button className={buttonStyle(styles.emailButton)}>Email</Button>
    </div>
  </section>
);

export default Contact;
