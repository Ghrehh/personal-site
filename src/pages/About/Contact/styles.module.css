.heading {
  color: var(--green);
  margin-bottom: 40px;
}

.buttonContainer {
  display: flex;
  justify-content: center;
}

.button {
  width: 150px;
  text-align: center;
  margin-right: 10px;
}

.button:last-child {
  margin-right: 0;
}

.gitHubButton {
  background-color: var(--green);
}

.twitterButton {
  background-color: var(--blue);
}

.emailButton {
  background-color: var(--red);
}
