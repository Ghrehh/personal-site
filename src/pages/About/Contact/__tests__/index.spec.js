import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';

import Contact from '../';

describe('Contact', () => {
  let testContext = {};

  beforeEach(() => {
    testContext.renderer = new ShallowRenderer();
    testContext.renderer.render(<Contact />);
    testContext.result = testContext.renderer.getRenderOutput();
  });

  it('renders the component correctly', () => {
    expect(testContext.result).toMatchSnapshot();
  });
});
