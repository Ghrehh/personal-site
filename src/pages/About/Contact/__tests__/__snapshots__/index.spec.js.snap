// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contact renders the component correctly 1`] = `
<section>
  <Heading
    className="heading"
  >
    CONTACT
  </Heading>
  <div
    className="buttonContainer"
  >
    <Button
      className="button twitterButton"
    >
      Twitter
    </Button>
    <Button
      className="button emailButton"
    >
      Email
    </Button>
  </div>
</section>
`;
